packages/utils/src/lib/report-to-md.ts

```diff
@@ -1,131 +1,309 @@
-import { NEW_LINE, details, headline, li, link, style, table } from './md/';
-import { CODE_PUSHUP_DOMAIN, FOOTER_PREFIX, countWeightedRefs } from './report';
-import { ScoredReport } from './scoring';
-import { reportHeadlineText, reportOverviewTableHeaders } from './utils';
+import {
+  AuditReport,
+  CategoryConfig,
+  Issue,
+  PluginReport,
+} from '@code-pushup/models';
+import {
+  NEW_LINE,
+  details,
+  h2,
+  h3,
+  headline,
+  li,
+  link,
+  style,
+  tableHtml,
+  tableMd,
+} from './md/';
+import {
+  FOOTER_PREFIX,
+  README_LINK,
+  countCategoryAudits,
+  formatDuration,
+  slugify,
+} from './report';
+import { EnrichedScoredAuditGroup, ScoredReport } from './scoring';
+import {
+  detailsTableHeaders,
+  formatReportScore,
+  getRoundScoreMarker,
+  getSeverityIcon,
+  getSquaredScoreMarker,
+  pluginMetaTableHeaders,
+  reportHeadlineText,
+  reportMetaTableHeaders,
+  reportOverviewTableHeaders,
+} from './utils';
 
 export function reportToMd(report: ScoredReport): string {
   // header section
   let md = reportToHeaderSection() + NEW_LINE;
 
-  // meta section
-  md += reportToMetaSection(report) + NEW_LINE + NEW_LINE;
-
   // overview section
   md += reportToOverviewSection(report) + NEW_LINE + NEW_LINE;
 
-  // details section
-  md += reportToDetailSection(report) + NEW_LINE;
+  // categories section
+  md += reportToCategoriesSection(report) + NEW_LINE + NEW_LINE;
+
+  // audits section
+  md += reportToAuditsSection(report) + NEW_LINE + NEW_LINE;
+
+  // about section
+  md += reportToAboutSection(report) + NEW_LINE + NEW_LINE;
 
   // footer section
-  md += `${FOOTER_PREFIX} ${link(CODE_PUSHUP_DOMAIN)}`;
+  md += `${FOOTER_PREFIX} ${link(README_LINK, 'Code PushUp')}`;
   return md;
 }
 
 function reportToHeaderSection(): string {
   return headline(reportHeadlineText) + NEW_LINE;
 }
 
-function reportToMetaSection(report: ScoredReport): string {
-  const { date, duration, version, packageName, plugins } = report;
+function reportToOverviewSection(report: ScoredReport): string {
+  const { categories } = report;
+  const tableContent: string[][] = [
+    reportOverviewTableHeaders,
+    ...categories.map(({ title, refs, score, slug }) => [
+      link(`#${slug}`, title),
+      `${getRoundScoreMarker(score)} ${style(formatReportScore(score))}`,
+      countCategoryAudits(refs, report.plugins).toString(),
+    ]),
```

Comment:
The category's `slug` might not correspond to the slugified `title`, this way should be less error-prone:

```suggestion
    ...categories.map(({ title, refs, score }) => [
      link(`#${slugify(title)}`, title),
      `${getRoundScoreMarker(score)} ${style(formatReportScore(score))}`,
      countCategoryAudits(refs, report.plugins).toString(),
    ]),
```

Reply:
Fixed